packages/next/src/build/webpack/loaders/next-flight-loader/action-proxy.ts

```diff
@@ -1,10 +1,24 @@
-export default function ensureServerEntryExports(actions: any[]) {
-  for (let i = 0; i < actions.length; i++) {
-    const action = actions[i]
-    if (typeof action !== 'function') {
-      throw new Error(
-        `A "use server" file can only export async functions, found ${typeof action}.`
-      )
+export default function createActionProxy(action: any, originalAction: any) {
+  action.bind = function (_: any, ...boundArgs: any[]) {
+    const newAction = async function (...args: any[]) {
+      if (originalAction) {
+        return originalAction(newAction.$$bound.concat(args))
+      } else {
+        // In this case we're calling the user-defined action directly.
+        return action(...newAction.$$bound, ...args)
+      }
```

Comment:
Might be better to just use the original native bind for this.

```
const bind = Function.prototype.bind;
...
const newAction = bind.apply(this, arguments);
...
```

Reply:
There's a bug that `Function` keyword is banned in the Edge runtime (makes sense to avoid `new Function("...")`, not sure if it's fixed now.